packages/wdio-sync/src/index.js

```diff
@@ -118,7 +118,7 @@ const runHook = function (hookFn, origFn, before, after, repeatTest = 0) {
             /**
              * user wants handle async command using promises, no need to wrap in fiber context
              */
-            if (hookFn.name === 'async') {
+            if (hookFn.constructor.name === 'AsyncFunction') {
                 return executeAsync.call(this, hookFn, repeatTest, filterSpecArgs(hookArgs))
```

Comment:
Is it possible that hookFn can be undefined or something without constructor property? 
In such case we'll get cannot read property name of undefined?
Do you think it makes sense to verify if constructor property exists in hookFn?

Reply:
sure updating the pr to check for constructor property on function prototype

Reply:
Do you think we still need to keep the old behavior? 
Is it possible at all to name function async? Isn't it a keyword? (Can't verify right now, just asking)

Reply:
it is possible to name fn as aync eventhough async is a keyword. Having an old workflow can be used as a fallback approach? Do you want me to remove old behavior?

Reply:
Not sure if anyone used it, we can keep it, I don't mind. Let's see if anyone else has some input of this.

I still think we can remove this one `typeof hookFn === 'function' `

Reply:
sure updating

Reply:
Thanks a lot!

Reply:
no problem